### PR TITLE
[efr32] add support for EFR32 MG13

### PR DESCRIPTION
**Pull request by garyButt**
This PR adds support for the Silabs EFR32 MG13 board.

### CODE REVIEW

File: examples/platforms/efr32mg13/README.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+```
+
+4. Build OpenThread Firmware (CLI example) on EFR32 platform.
+```bash
+$ cd <path-to-openthread>
+$ ./bootstrap
+```
+For EFR32MG13™ Mighty Gecko Wireless Starter Kit:
+```bash
+$ make -f examples/Makefile-efr32mg13 BOARD=BRD4161A
```
Comment (jwhui) [inline examples/platforms/efr32mg13/README.md]: The BRD4151A does not seem to be supported for efr32mg13. Did you mean BRD4168A?

File: examples/platforms/efr32mg13/README.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+```
+or alternatively for the Thunderboard™ Sense 2:
+
+```bash
+$ make -f examples/Makefile-efr32mg13 BOARD=BRD4166A
```
Comment (jwhui) [inline examples/platforms/efr32mg13/README.md]: BRD4166A does not seem to be supported for efr32mg13. Is this example for Thunderboard Sense 2 valid for the efr32mg13?

File: examples/platforms/efr32mg13/README.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+[CLI]: https://github.com/openthread/openthread/blob/master/src/cli/README.md
+
+## Verification
+
+The following toolchain has been used for testing and verification:
+   - gcc version 7.3.1
+
+The EFR32 example has been verified with following Flex SDK/RAIL Library version:
+   - Flex SDK version 2.6.0.0
```
Comment (jwhui) [inline examples/platforms/efr32mg13/README.md]: Should this be v2.7?

File: examples/platforms/efr32mg13/README.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
+# OpenThread on EFR32MG13 Example
+
+This directory contains example platform drivers for the [Silicon Labs EFR32MG13][efr32mg13]
+based on [EFR32™ Mighty Gecko Wireless Starter Kit][SLWSTK6000B] or [Thunderboard™ Sense 2 Sensor-to-Cloud Advanced IoT Development Kit][SLTB004A].
```
Comment (jwhui) [inline examples/platforms/efr32mg13/README.md]: Is Thunderboard Sense 2 valid for efr32mg13?

File: examples/platforms/efr32mg13/efr32mg13.ld [hunk 5]
```diff
@@ -0,0 +1,229 @@
+/***************************************************************************//**
```
Comment (jwhui) [inline examples/platforms/efr32mg13/efr32mg13.ld]: A similar linker script was not needed in `examples/platforms/efr32mg12` and `examples/platforms/efr32mg21`. Is something different with the efr32mg13 that requires this linker script to be here?

If so, we'll either need to put the OpenThread license header on top or move this into `third_party/silabs`.

File: examples/platforms/efr32mg13/sleepy-demo/README.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+For setting up the build environment refer to [examples/platforms/efr32mg13/README.md](../README.md).
+
+
+## 1. Build
+
+```bash
+$ cd <path-to-openthread>
+$ ./bootstrap
+$ make -f examples/Makefile-efr32mg13 COMMISSIONER=1 JOINER=1 DHCP6_CLIENT=1 DHCP6_SERVER=1 BOARD=BRD4161A
```
Comment (jwhui) [inline examples/platforms/efr32mg13/sleepy-demo/README.md]: Use `BOARD=BRD4168A`?

File: examples/platforms/efr32mg13/README.md (hunk 1 above)
Comment (garyButt) [reply examples/platforms/efr32mg13/README.md]: Yes, should be 4168A. Fixed.

File: examples/platforms/efr32mg13/README.md (hunk 2 above)
Comment (garyButt) [reply examples/platforms/efr32mg13/README.md]: Thunderboard Sense is of type mg12, so should not be mentoined in mg13 readme.

File: examples/platforms/efr32mg13/README.md (hunk 3 above)
Comment (garyButt) [reply examples/platforms/efr32mg13/README.md]: Fixed

File: examples/platforms/efr32mg13/efr32mg13.ld (hunk 5 above)
Comment (garyButt) [reply examples/platforms/efr32mg13/efr32mg13.ld]: mg13 linker script file deleted (modified linker scripts will be required for all efr32 platforms when we merge in the nvm3 pr at a later date, but required for the mg13 pr)

File: examples/platforms/efr32mg13/sleepy-demo/README.md (hunk 6 above)
Comment (garyButt) [reply examples/platforms/efr32mg13/sleepy-demo/README.md]: Fixed.

**Pull request merged**